### PR TITLE
Application-level tagging

### PR DESCRIPTION
**Pull request by ThatJoeMoore**
This PR will aim to make it easier to apply tags across all resources in a Handelfile.

For the BYU use case, this makes it easier to have consistent tagging with required tags like 'team' and 'data-sensitivity'.

This gets us part of the way towards solving the issues raised in #332.

### CODE REVIEW

**Comment by ThatJoeMoore** [comment]
So far, this is just the docs.  I wanted to get the feedback loop going while I start implementing.

Pinging @dsw88

**Comment by dsw88** [comment]
This looks great! It's definitely a needed feature to reduce duplication for most tagging situations.

I do have one question that's not in the docs yet. When there is both a top-level tag and a resource-level tag with the same name, which one wins? I definitely prefer the resource-level tags overriding any top-level tags, and I imagine that is probably what you are expecting too.

**Comment by ThatJoeMoore** [comment]
That's what I'd expect.  My implementation plan is to apply the application-level tags to the cloudformation stacks. The resource tags would remain on the individual resources, and in CFN, I'm pretty sure that resource tags override stack tags.

**Comment by dsw88** [comment]
I've been going with the pattern that tags are applied to both the stack as well as the applicable resources inside the stack. I was assuming global tags would follow this same pattern where they are applied to every stack as well as the resources inside them. What do you think about continuing that pattern?

**Comment by ThatJoeMoore** [comment]
This is now ready for review.  I ended up piggybacking on the existing tag code, so I believe that the tags will get propagated everywhere they were before, just like 'app' and 'env'.

I'm not sure we need to propagate all tags to all resources inside of a CFN stack, because CFN auto-tags everything in the stack with the stack tags (if not overwritten by the resource tags).

**Comment by ThatJoeMoore** [comment]
Fun fact: with this PR, we have an even 500 test cases in our suite ✋ 💯

File: src/common/deploy-phase-common.ts [hunk 1]
```diff
@@ -192,16 +192,17 @@ export function getResourceName(serviceContext: ServiceContext<ServiceConfig>) {
...
-    if (serviceParams.tags) {
-        tags = Object.assign(tags, serviceParams.tags);
-    }
+    const appTags = serviceContext.tags;
+
+    const serviceTags = serviceParams.tags;
 
-    return tags;
+    // Service tags overwrite app tags, which overwrite Handel tags
+    return Object.assign({}, handelTags, appTags, serviceTags);
```
Comment (dsw88) [inline src/common/deploy-phase-common.ts]: I was about to ask why you were adding these when you haven't checked that they exist, but then I went and tried it out and it turns out Object.assign handles undefined values without a problem. You learn something new every day!

Comment (ThatJoeMoore) [reply src/common/deploy-phase-common.ts]: Yup!  That was my TIL from a couple weeks ago :D

**Comment by dsw88** [comment]
Thanks for implementing this @ThatJoeMoore!

**Pull request merged**